src/zkp/pifac.rs

```diff
@@ -24,43 +33,62 @@ use std::fmt::Debug;
 use tracing::error;
 use zeroize::ZeroizeOnDrop;
 
+/// Proof that the modulus N can be factored into two numbers greater than 2^ell
+/// for a parameter ell.
 #[derive(Debug, Serialize, Deserialize, Clone)]
 pub(crate) struct PiFacProof {
```

Comment:
todo: In all of hte documentation here, I would remove references to intermediate variables that aren't exposed in the proof (`x`, `y`, `meu`, `neu`, `alpha`, `beta`, etc). Those won't make sense unless someone is looking at the paper directly. 

When the intermediate var is commitment randomness I think you can just not talk about it. A commitment always has commitment randomness but nobody needs to know about it explicitly except in the inline implementation.

When the intermediate var is a mask I think it's fine to also ignore it in the doc and just talk about the thing that is getting masked without naming the mask itself.

Reply:
@marsella  But then explaining more things is better no? In some sense, people who understands the paper and looks at the code has better visibility since the var names used in the code and the paper are different and if I completely remove all references to alpha, x, etc. then it would be hard for people to understand and makes the documentation incomplete in my opinion. Let me know your thoughts. If you think it confuses people more, then I would completely get rid of them.

Reply:
This section should aim to describe the overall structure of the proof, via good names, good types, and succinct docs:
- what are we sending?
- what's the type of each field?
- what's the role of each field in the proof?

I agree that we should try to reference all the variables generated as part of the proof, but I think the details about, say, names of commitment randomness belong inline in the proof -- where we actually create them. Right now many of the names in `prove()` do map directly to the paper, so auditors can tell what's what. There are a few that have gotten more descriptive names (like `p_mask` instead of `alpha`); I think it would be reasonable to add docs at the place where we create those to map to the paper name. 

Does that sound reasonable to you? The goal is that someone looking at this `PiFacProof` type who's familiar with how sigma proofs work will be able to scan it quickly and understand the general shape without getting bogged down in implementation details. If they want those details, they can look at the `prove` and `verify` methods.

Reply:
Oh, I was imagining a bit more of a middle ground than this. E.g. for the `*_mask_commitment`s I would say
```
/// Commitment to a mask for p (`A` in the paper)
```

and for the `w1/w2`s, something like:
```
/// Masked commitment randomness used to form `p_commitment`.
```

Reply:
Please update the docs for `{p, q}_mask_commitment` and `masked_{p, q}_commitment_randomness` per comment above.